### PR TITLE
Feature/sphinx docs

### PR DESCRIPTION
**Pull request by billbsing**
Adding Sphinx doc generation.
To build 
```bash
cd docs
./build.sh
```
The above text is also in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,7 +72,15 @@ Python 3.6
...
     python3 setup.py test
     ```
+## Documentation
 
+1. Build Sphinx documentation
+    To build the Syphinx auto documentation, you need to do the following:
+    ```bash
+    cd docs
+    ./build.sh
+    ```
```
Comment (tmarble) [inline README.md]: in the top level Makefile how about 'make docs'

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+SPHINXPROJ    = squid-py
```
Comment (tmarble) [inline docs/Makefile]: Is this needed?

File: docs/build.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
+make html
```
Comment (tmarble) [inline docs/build.sh]: This won't be necessary when we have a top level make docs target

File: docs/Makefile (hunk 2 above)
Comment (bangtree) [reply docs/Makefile]: I'd say yes. In Fedora, at least F28, the _sphinx-build_ command is _sphinx-build-3_ , so having an easy override is handy.

**Comment by bangtree** [comment]
LGTM. I've built the docs locally using this and they look good.

**Pull request merged**